src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,154 @@
+    public static final int BELOW_MINIMUM_AGE = 17;
+    public static final int MINIMUM_AGE = 18;
+    public static final int OVER_MINIMUM_AGE = 21;
+    public static final int EXPECTED_TWO = 2;
+    public static final int EXPECTED_FOUR = 4;
+    public static final int FIRST = 0;
+    public static final String LOGIN_YURI = "mylogin";
+    public static final String LOGIN_DASHA = "DZlogin";
+    public static final String LOGIN_NAZAR = "nazar2005";
+    public static final String CORRECT_LOGIN = "qwerty";
+    public static final String INCORRECT_LOGIN = "log";
+    public static final String CORRECT_LOGIN_SIX_LETTERS = "mlogin";
+    public static final String CORRECT_PASSWORD = "qwertyu";
+    public static final String CORRECT_PASSWORD_SIX_LETTERS = "123456";
+    public static final String INCORRECT_PASSWORD = "qwer";
+    public static final String EMPTY_LINE = "";
+
+    private static RegistrationService registrationService;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @AfterEach
+    void tearDown() {
+        people.clear();
+    }
+
+    @Test
+    void register_nullLoginValue_NotOk() {
+        assertThrows(RegistrationException.class, () -> registrationService.register(
+                new User(null, CORRECT_PASSWORD, MINIMUM_AGE)));
+    }
+
+    @Test
+    void register_NullPasswordValue_NotOk() {
+        assertThrows(RegistrationException.class, () -> registrationService.register(
+                new User(LOGIN_YURI, null, MINIMUM_AGE)));
+    }
+
+    @Test
+    void register_NullAgeValue_NotOk() {
+        assertThrows(RegistrationException.class, () -> registrationService.register(
+                new User(LOGIN_YURI, CORRECT_PASSWORD, null)));
+    }
+
+    @Test
+    void register_correctEge_Ok() {
+        registrationService.register(new User(LOGIN_YURI, CORRECT_PASSWORD, MINIMUM_AGE));
+        registrationService.register(new User(LOGIN_DASHA, CORRECT_PASSWORD, OVER_MINIMUM_AGE));
+        assertTrue(people.contains(new User(LOGIN_YURI, CORRECT_PASSWORD, MINIMUM_AGE)));
+        assertTrue(people.contains(new User(LOGIN_DASHA, CORRECT_PASSWORD, OVER_MINIMUM_AGE)));
+        assertEquals(EXPECTED_TWO, people.size());
+    }
+
+    @Test
+    void register_negativeAge_NotOk() {
+        assertThrows(RegistrationException.class, () -> registrationService.register(
+                new User(CORRECT_LOGIN, CORRECT_PASSWORD, NEGATIVE_AGE)));
+    }
+
+    @Test
+    void register_zeroAge_NotOk() {
+        assertThrows(RegistrationException.class, () -> registrationService.register(
+                new User(CORRECT_LOGIN, CORRECT_PASSWORD, ZERO_AGE)));
+    }
+
+    @Test
+    void register_belowMinimumAge_NotOk() {
+        assertThrows(RegistrationException.class, () -> registrationService.register(
+                new User(CORRECT_LOGIN, CORRECT_PASSWORD, BELOW_MINIMUM_AGE)));
+    }
+
+    @Test
+    void register_correctLogin_Ok() {
```

Comment:
Similar to the previous comment, each user registration should be tested in its own test method to ensure independence. The test method name should also reflect the specific scenario being tested.